ffi/Makefile.linux

```diff
@@ -4,22 +4,27 @@ CXX ?= g++
 # -flto and --exclude-libs allow us to remove those parts of LLVM we don't use
 CXX_FLTO_FLAGS ?= -flto
 LD_FLTO_FLAGS ?= -flto -Wl,--exclude-libs=ALL
+# -fPIC is required when compiling objects for a shared library
+CXX_FPIC_FLAGS ?= -fPIC
 
-CXXFLAGS := $(CPPFLAGS) $(CXXFLAGS) $(LLVM_CXXFLAGS) $(CXX_FLTO_FLAGS)
+CXXFLAGS := $(CPPFLAGS) $(CXXFLAGS) $(LLVM_CXXFLAGS) $(CXX_FLTO_FLAGS) $(CXX_FPIC_FLAGS)
 LDFLAGS := $(LDFLAGS) $(LLVM_LDFLAGS) $(LD_FLTO_FLAGS)
 LIBS = $(LLVM_LIBS)
 INCLUDE = core.h
-SRC = assembly.cpp bitcode.cpp core.cpp initfini.cpp module.cpp value.cpp \
-	  executionengine.cpp transforms.cpp passmanagers.cpp targets.cpp dylib.cpp \
-	  linker.cpp object_file.cpp custom_passes.cpp
+OBJ = assembly.o bitcode.o core.o initfini.o module.o value.o \
+	  executionengine.o transforms.o passmanagers.o targets.o dylib.o \
+	  linker.o object_file.o custom_passes.o
 OUTPUT = libllvmlite.so
 
 all: $(OUTPUT)
 
-$(OUTPUT): $(SRC) $(INCLUDE)
+.cpp.o: $(INCLUDE)
```

Comment:
What are these lines doing? If I comment them out, I still seem to get the exact same invocation of `g++` for compiling, and the same `.so` output. (My makefile expertise is limited - is there some subtlety I'm missing here?

Reply:
They define how to create an `.o` file from `.cpp`.

You get the same result most likely because GNU make has an implicit rule for building C++ files. However, POSIX does not specify such a rule (for C++, it does for plain C), so other make implementations may be missing one.

Reply:
Many thanks for the explanation!